lib/web_ui/lib/src/engine/font_fallbacks.dart

```diff
@@ -257,56 +265,49 @@ class FontFallbackManager {
       }
     }
 
-    // If the list of best fonts are all CJK fonts, choose the best one based
-    // on locale. Otherwise just choose the first font.
+    NotoFont? bestFontForLanguage;
     if (bestFonts.length > 1) {
+      // If the list of best fonts are all CJK fonts, choose the best one based
+      // on user preferred language. Otherwise just choose the first font.
       if (bestFonts.every((NotoFont font) =>
-          font == _notoSansSC ||
-          font == _notoSansTC ||
-          font == _notoSansHK ||
-          font == _notoSansJP ||
-          font == _notoSansKR)) {
-        final String language = domWindow.navigator.language;
-
-        if (language == 'zh-Hans' ||
-            language == 'zh-CN' ||
-            language == 'zh-SG' ||
-            language == 'zh-MY') {
-          if (bestFonts.contains(_notoSansSC)) {
-            bestFont = _notoSansSC;
-          }
-        } else if (language == 'zh-Hant' ||
-            language == 'zh-TW' ||
-            language == 'zh-MO') {
-          if (bestFonts.contains(_notoSansTC)) {
-            bestFont = _notoSansTC;
-          }
-        } else if (language == 'zh-HK') {
-          if (bestFonts.contains(_notoSansHK)) {
-            bestFont = _notoSansHK;
-          }
-        } else if (language == 'ja') {
-          if (bestFonts.contains(_notoSansJP)) {
-            bestFont = _notoSansJP;
-          }
-        } else if (language == 'ko') {
-          if (bestFonts.contains(_notoSansKR)) {
-            bestFont = _notoSansKR;
-          }
-        } else if (bestFonts.contains(_notoSansSC)) {
-          bestFont = _notoSansSC;
+          _isNotoSansSC(font) ||
+          _isNotoSansTC(font) ||
+          _isNotoSansHK(font) ||
+          _isNotoSansJP(font) ||
+          _isNotoSansKR(font))) {
+        if (_language == 'zh-Hans' ||
```

Comment:
Can `_language` ever be just `'zh'`?

Reply:
Not sure. Revising the logic wasn't a goal for this PR. I was just trying to make the logic work the same way after the font split.

That said, I think you're right in the other comment that the `zh` case is handled later by the catch-all.